source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -143,6 +145,36 @@ namespace
         }
     }
 
+    void readUiPosition(const ElementPtr src, PvtObject* dest)
```

Comment:
I know this is the way that MaterialX typically does things, but is const ElementPtr& src better? This is because you're not sharing ownership of the object, just using it.

Reply:
Good point, then we can avoid the extra ref pointer increment/decrement. Will fix it!